packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -143,8 +150,8 @@ async function registerTsLoader(loader: TsLoaderModule): Promise<TsLoader> {
         moduleTypes: {
           '**': 'cjs',
         },
-        transpileOnly:
-          process.env.JEST_CONFIG_TRANSPILE_ONLY?.toLowerCase() === 'true',
```

Comment:
removed this env variable which was added in #15161, we can now pass this via docblock comment